### PR TITLE
Issue1977 - Add support for recursive deletion of entities

### PR DESCRIPTION
**Pull request by sureshms**
### Describe your changes :

See #1960 and #1977 for more details

#
### Type of change :

- [] Bug fix
- [] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach

### CODE REVIEW

File: bootstrap/sql/mysql/v003__create_db_connection_info.sql [hunk 1]
```diff
@@ -106,6 +106,10 @@ ADD INDEX (deleted);
 UPDATE entity_relationship
 SET relation = 10 WHERE fromEntity = 'team' AND toEntity = 'user' AND relation = 0;
 
+-- Change "dashboard -- contains --> chart" relationship to "dashboard -- has --> chart" relationship
```
Comment (mithmatt) [inline bootstrap/sql/mysql/v003__create_db_connection_info.sql:109]: Should this go into v004?

**Review by mithmatt — APPROVED** [review]

File: bootstrap/sql/mysql/v003__create_db_connection_info.sql (hunk 1 above)
Comment (harshach) [reply bootstrap/sql/mysql/v003__create_db_connection_info.sql:109]: It’s not needed. Till 0.7 we have v002 version https://github.com/open-metadata/OpenMetadata/tree/0.7.0/bootstrap/sql/mysql . When users upgrade the bootstrap migrate will apply v003 . So each release if we have any schema changes we can increment this by one. All changes related to that version can belong to one sql file

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/exception/CatalogGenericExceptionMapper.java [hunk 2]
```diff
@@ -42,6 +42,7 @@ public CatalogGenericExceptionMapper() {}
 
   @Override
   public Response toResponse(Throwable ex) {
+    ex.printStackTrace();
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/exception/CatalogGenericExceptionMapper.java:45]: Turn this into log or do we intend to return full stack trace here?

**Review by harshach — APPROVED** [review]
Left a minor nit on the ex.printStackTrace()

File: bootstrap/sql/mysql/v003__create_db_connection_info.sql (hunk 1 above)
Comment (sureshms) [reply bootstrap/sql/mysql/v003__create_db_connection_info.sql:109]: Yes to what @harshach said. I deleted v004 for that reason. We need to pay attention and have one version file per release, if database needs changes.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/exception/CatalogGenericExceptionMapper.java (hunk 2 above)
Comment (sureshms) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/exception/CatalogGenericExceptionMapper.java]: Deleted the line used for debugging

**Pull request merged**